test/fs.test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const path = require('path');
+const { createFsFromVolume, Volume } = require('memfs');
+const webpack = require('webpack');
+const Server = require('../lib/Server');
+const config = require('./fixtures/simple-config/webpack.config');
+
+describe('Validation', () => {
+  let compiler;
+  let server;
+
+  beforeAll(() => {
+    compiler = webpack(config);
+  });
+
+  describe('fs', () => {
+    let memfs;
+    beforeEach(() => {
+      memfs = createFsFromVolume(new Volume());
+      // We need to patch memfs
+      // https://github.com/webpack/webpack-dev-middleware#fs
+      memfs.join = path.join;
+    });
+    afterEach((done) => {
+      server.close(() => {
+        done();
+      });
+    });
+
+    it('should allow fs to be memfs', () => {
+      try {
+        // eslint-disable-next-line no-new
+        server = new Server(compiler, { fs: memfs });
+      } catch (err) {
+        if (err === 'ValidationError') {
+          throw err;
+        }
+
+        throw new Error("Validation failed and it shouldn't");
```

Comment:
Each test should contain `expect`, let's move `error` (`let error = null`) on top and check `expect(error).toBe(null)`